### PR TITLE
v3.0, dev/reference: add tidb-operator-overview

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? 

+ add `tidb-operator-overview.md`
+ update `architecture.md` to add TiDB Operator-related content
+ update `contribute.md` to add TiDB Operator-related content

### What is the related PR or file link(s)? 

https://github.com/pingcap/docs-cn/pull/1625

### Which version does your change affect? 

v3.0, dev

### Checklist 

- [ ] Add a new file to `TOC.md`
- [ ] No Tab spaces anywhere 
- [ ] Leave a blank line both before and after a code block
- [ ] Keep the first level heading consistent with `title` in metadata
- [ ] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

File: dev/reference/tidb-operator-overview.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+TiDB Operator is an automatic operation and maintenance system for TiDB cluster in Kubernetes that provides a full life-cycle management for TiDB including deployment, upgrades, expansion, backup, recovery, and configuration changes. With TiDB Operator, TiDB can run seamlessly in the Kubernetes clusters deployed on public or private cloud.
+
+## Architecture of TiDB Operator
+
+![TiDB Operator Overview](/media/tidb-operator-overview.png)
+
+TiDB Operator consists of the custom resources `TidbCluster`, the `tidb-controller-manager` components, the `tidb-scheduler` components and the command line tools `tkctl`:
+
+* `TidbCluster` is a custom resource defined using CRD (`CustomResourceDefinition`) to describe the expected state of the TiDB cluster;
```
Comment (aylei) [inline dev/reference/tidb-operator-overview.md]: In Kubernetes community, the idiomatic phrase is "desired state"

File: dev/reference/tidb-operator-overview.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+* `tidb-controller-manager` is a set of custom controllers in Kubernetes. These controllers constantly compare the expected state recorded in the `TidbCluster` object with the actual state of the TiDB cluster. They adjust the resources in Kubernetes to drive the TiDB cluster to meet the desired state;
```
Comment (aylei) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
* `tidb-controller-manager` is a set of custom controllers in Kubernetes. These controllers constantly compare the desired state recorded in the `TidbCluster` object with the actual state of the TiDB cluster, then adjust the resources in Kubernetes to drive the TiDB cluster to meet the desired state.
```

File: dev/reference/tidb-operator-overview.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+* `tidb-scheduler` is an extension of the Kubernetes scheduler that injects the scheduling logic specific to TiBD clusters for the Kubernetes scheduler;
```
Comment (aylei) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
* `tidb-scheduler` is a Kubernetes scheduler extender that injects the TiDB specific scheduling policies for the Kubernetes scheduler;
```

`scheduler extender` is a Kubernetes terminology, and it is more idiomatic to say "schedule policy" or "schedule strategy"

File: dev/reference/tidb-operator-overview.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+* `tkctl` is the command line interface for the TiDB clusters in Kubernetes. It is used for issues related to the operation, maintenance and diagnosis of these clusters.
+
+![TiDB Operator Control Flow](/media/tidb-operator-control-flow.png)
+
+The diagram above is the analysis of the control flow of TiDB Operator. Because TiDB clusters also need components such as monitoring, initialization, scheduled backup, Binlog and so on, the definition of these clusters are encapsulated in a TiDB Operator using Helm Chart. The overall control process is as follows:
```
Comment (aylei) [inline dev/reference/tidb-operator-overview.md]: "TiDB Operator encapsulates the definition of these components in the Helm chart"  maybe better

File: dev/reference/tidb-operator-overview.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+1. The user creates a `TidbCluster` object and a corresponding series of Kubernetes-native objects through Helm, such as `CronJob` that performs scheduled backups;
```
Comment (aylei) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
1. The user creates a `TidbCluster` object and a corresponding series of Kubernetes-native objects through Helm, such as a `CronJob` that performs scheduled backups;
```

File: dev/reference/tidb-operator-overview.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+2. TiDB Operator will watch `TidbCluster` and other related objects, and constantly adjust the `StatefulSet` and `Service` objects of PD, TiKV, TiDB based on the actual state of the cluster;
+3. Kubernetes' native controller creates, updates or deletes the corresponding `Pod` based on the objects such as `StatefulSet`, `Deployment`, `CronJob` and so on;
```
Comment (aylei) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
3. Kubernetes' native controllers create, update and delete the corresponding `Pod` based on the objects such as `StatefulSet`, `Deployment`, `CronJob` and so on;
```

File: dev/reference/tidb-operator-overview.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+4. In the `Pod` declaration of PD, TiKV, TiDB, the `tidb-scheduler` scheduler is specified which applies the specific scheduling logic of TiDB when scheduling the corresponding `Pod`.
+
+Based on the above declarative control flow, TiDB Operator automatically performs health check and fault recovery for the cluster nodes. Operations such as deployment, upgrade and scaling can also be completed by modifying the `TidbCluster` object declaration by "one click".
```
Comment (aylei) [inline dev/reference/tidb-operator-overview.md]: "by ... by ..." sounds weird, can we come up with a better expression?

**Comment by aylei** [comment]
@tennix @gregwebs PTAL

File: dev/reference/tidb-operator-overview.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
+---
+title: Overview of TiDB Operator
+summary: Learn the overview of TiDB Operator.
+category: reference
+---
+
+# Overview of TiDB Operator
+
+TiDB Operator is an automatic operation and maintenance system for TiDB cluster in Kubernetes that provides a full life-cycle management for TiDB including deployment, upgrades, scaling, backup, fail-over, and configuration changes. With TiDB Operator, TiDB can run seamlessly in the Kubernetes clusters deployed on public or private cloud.
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
TiDB Operator is an automatic operation and maintenance system for TiDB clusters in Kubernetes. It provides a full management life-cycle for TiDB including deployment, upgrades, scaling, backup, fail-over, and configuration changes. With TiDB Operator, TiDB can run seamlessly in the Kubernetes clusters deployed on a public or private cloud.
```

File: dev/reference/tidb-operator-overview.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+## Architecture of TiDB Operator
+
+![TiDB Operator Overview](/media/tidb-operator-overview.png)
+
+TiDB Operator consists of the custom resources `TidbCluster`, the `tidb-controller-manager` components, the `tidb-scheduler` components and the command line tools `tkctl`:
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
TiDB Operator consists of the following components:
```

File: dev/reference/tidb-operator-overview.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+* `TidbCluster` is a custom resource defined using CRD (`CustomResourceDefinition`) to describe the desired state of the TiDB cluster;
+* `tidb-controller-manager` is a set of custom controllers in Kubernetes. These controllers constantly compare the desired state recorded in the `TidbCluster` object with the actual state of the TiDB cluster. They adjust the resources in Kubernetes to drive the TiDB cluster to meet the desired state;
+* `tidb-scheduler` is a Kubernetes scheduler extender that injects the TiDB specific scheduling policies for the Kubernetes scheduler;
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
* `tidb-scheduler` is a Kubernetes scheduler extension that injects TiDB specific scheduling policies to the Kubernetes scheduler;
```

File: dev/reference/tidb-operator-overview.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+* `tkctl` is the command line interface for the TiDB clusters in Kubernetes. It is used for issues related to the operation, maintenance and diagnosis of these clusters.
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
* `tkctl` is the command-line interface for TiDB clusters in Kubernetes. It is used for cluster maintenance operations and troubleshooting cluster issues.
```

File: dev/reference/tidb-operator-overview.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+![TiDB Operator Control Flow](/media/tidb-operator-control-flow.png)
+
+The diagram above is the analysis of the control flow of TiDB Operator. Because TiDB clusters also need components such as monitoring, initialization, scheduled backup, Binlog and so on, TiDB Operator encapsulates the definition of these components in the Helm chart. The overall control process is as follows:
+
+1. The user creates a `TidbCluster` object and a corresponding series of Kubernetes-native objects through Helm, such as a `CronJob` that performs scheduled backups;
+2. TiDB Operator will watch `TidbCluster` and other related objects, and constantly adjust the `StatefulSet` and `Service` objects of PD, TiKV, TiDB based on the actual state of the cluster;
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
2. TiDB Operator watches `TidbCluster` and other related objects, and constantly adjust the `StatefulSet` and `Service` objects of PD, TiKV, and TiDB based on the actual state of the cluster;
```

File: dev/reference/tidb-operator-overview.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+3. Kubernetes' native controller creates, updates and deletes the corresponding `Pod` based on the objects such as `StatefulSet`, `Deployment`, `CronJob` and so on;
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
3. Kubernetes' native controller creates, updates, and deletes the corresponding `Pod` based on objects such as `StatefulSet`, `Deployment`, and `CronJob`;
```

File: dev/reference/tidb-operator-overview.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+4. In the `Pod` declaration of PD, TiKV, TiDB, the `tidb-scheduler` scheduler is specified which applies the specific scheduling logic of TiDB when scheduling the corresponding `Pod`.
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
4. In the `Pod` declaration of PD, TiKV, and TiDB, the `tidb-scheduler` scheduler is specified. `tidb-scheduler` applies the specific scheduling logic of TiDB when scheduling the corresponding `Pod`.
```

File: dev/reference/tidb-operator-overview.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+
+TiDB Operator provides several ways to deploy TiDB clusters in Kubernetes:
+
++ For test environment:
+    - [DinD](tidb-in-kubernetes/get-started/deploy-tidb-from-kubernetes-dind.md): Deploy TiDB clusters in a local DinD environment using TiDB Operator
+    - [Minikube](tidb-in-kubernetes/get-started/deploy-tidb-from-kubernetes-minikube.md): Deploy TiDB clusters in a local Minikube environment using TiDB Operator
+    - [GKE](tidb-in-kubernetes/get-started/deploy-tidb-from-kubernetes-gke.md): Deploy TiDB clusters on GKE using TiDB Operator
+
++ For production environment:
+    - On public cloud: see the [Deploy TiDB on AWS EKS](tidb-in-kubernetes/deploy/aws-eks.md), [Deploy TiDB on GCP GKE](tidb-in-kubernetes/deploy/gcp-gke.md) or [Deploy TiDB on Alibaba Cloud Kubernetes](tidb-in-kubernetes/deploy/alibaba-cloud.md) to deploy TiDB clusters for a production environment on the specific public cloud and perform subsequent operation and maintenance;
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
    - On public cloud: see [Deploy TiDB on AWS EKS](tidb-in-kubernetes/deploy/aws-eks.md), [Deploy TiDB on GCP GKE](tidb-in-kubernetes/deploy/gcp-gke.md), or [Deploy TiDB on Alibaba Cloud Kubernetes](tidb-in-kubernetes/deploy/alibaba-cloud.md) to deploy TiDB clusters for a production environment on a specific public cloud and perform subsequent operation and maintenance tasks;
```

File: dev/reference/tidb-operator-overview.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+    - In the existing Kubernetes cluster: first install the TiDB Operator in a cluster according to [Deploy TiDB Operator in Kubernetes](tidb-in-kubernetes/deploy/tidb-operator.md), then deploy your TiDB clusters according to [TiDB in General Kubernetes](tidb-in-kubernetes/deploy/general-kubernetes.md). You alo need to adjust the configuration of the Kubernetes cluster according to [Prerequisites for TiDB in Kubernetes](tidb-in-kubernetes/deploy/prerequisites.md) and configure the local PV for your Kubernetes cluster to meet the need of low latency of local storage on TiKV according to [Local PV Configuration](tidb-in-kubernetes/reference/configuration/local-pv.md).
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
    - In an existing Kubernetes cluster: first install TiDB Operator in a Kubernetes cluster according to [Deploy TiDB Operator in Kubernetes](tidb-in-kubernetes/deploy/tidb-operator.md), then deploy your TiDB clusters according to [TiDB in General Kubernetes](tidb-in-kubernetes/deploy/general-kubernetes.md). You also need to adjust the configuration of the Kubernetes cluster based on [Prerequisites for TiDB in Kubernetes](tidb-in-kubernetes/deploy/prerequisites.md) and configure the local PV for your Kubernetes cluster to achieve low latency of local storage for TiKV according to [Local PV Configuration](tidb-in-kubernetes/reference/configuration/local-pv.md).
```

File: dev/reference/tidb-operator-overview.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+Before deploying TiDB on either of the above two environments, you can always see [TiDB Cluster Configuration Document](/tidb-in-kubernetes/reference/configuration/tidb-cluster.md) to customize TiDB configurations.
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
Before deploying TiDB on any of the above two environments, you can always refer to [TiDB Cluster Configuration Document](/tidb-in-kubernetes/reference/configuration/tidb-cluster.md) to customize TiDB configurations.
```

File: dev/reference/tidb-operator-overview.md [hunk 18]
```diff
@@ -0,0 +1,70 @@
...
+
+After the deployment is complete, see the following documents to use, operate and maintain the TiDB cluster in Kubernetes:
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
After the deployment is complete, see the following documents to use, operate, and maintain TiDB clusters in Kubernetes:
```

File: dev/reference/tidb-operator-overview.md [hunk 19]
```diff
@@ -0,0 +1,70 @@
...
++ [Automatic Failover](tidb-in-kubernetes/maitain/auto-failover.md)
++ [Monitor a TiDB Cluster in Kubernetes](tidb-in-kubernetes/monitor/tidb-in-kubernetes.md)
++ [Collect TiDB Logs in Kubernetes](tidb-in-kubernetes/maitain/log-collecting.md)
+
+When a problem occurs and the cluster needs diagnosis, you can:
+
++ See [TiDB FAQs in Kubernetes](tidb-in-kubernetes/faq.md) for any available solution;
++ See [Troubleshoot TiDB in Kubernetes](tidb-in-kubernetes/troubleshoot.md) to shoot troubles.
+
+TiDB in Kubernetes provides a dedicated command line tool `tkctl` for cluster management and auxiliary diagnostics. Meanwhile, some of TiDB's eco-tools are used differently in Kubernetes. You can:
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
TiDB in Kubernetes provides a dedicated command-line tool `tkctl` for cluster management and auxiliary diagnostics. Meanwhile, some of TiDB's tools are used differently in Kubernetes. You can:
```

File: dev/reference/tidb-operator-overview.md [hunk 20]
```diff
@@ -0,0 +1,70 @@
...
+
++ Use `tkctl` according to [`tkctl` Guide](tidb-in-kubernetes/reference/tools/tkctl.md );
++ See [Tools in Kubernetes](tidb-in-kubernetes/reference/tools/in-kubernetes.md) to understand how TiDB eco-tools are used in Kubernetes.
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
+ See [Tools in Kubernetes](tidb-in-kubernetes/reference/tools/in-kubernetes.md) to understand how TiDB tools are used in Kubernetes.
```

File: dev/reference/tidb-operator-overview.md [hunk 21]
```diff
@@ -0,0 +1,70 @@
...
+
+Finally, when a new version of TiDB Operator is released, you can refer to [Upgrade TiDB Operator](tidb-in-kubernetes/upgrade/tidb-operator.md) for version updates.
```
Comment (dcalvin) [inline dev/reference/tidb-operator-overview.md]:
```suggestion
Finally, when a new version of TiDB Operator is released, you can refer to [Upgrade TiDB Operator](tidb-in-kubernetes/upgrade/tidb-operator.md) to upgrade to the latest version.
```

**Comment by TomShawn** [comment]
PTAL again, thanks! @dcalvin @aylei

**Comment by lilin90** [comment]
@TomShawn Some more updates are made in https://github.com/pingcap/docs-cn/pull/1625. Please help update the English version accordingly. Thanks!

**Comment by TomShawn** [comment]
> @TomShawn Some more updates are made in [pingcap/docs-cn#1625](https://github.com/pingcap/docs-cn/pull/1625). Please help update the English version accordingly. Thanks!

Sure!

**Pull request merged**